### PR TITLE
Revamp configs and more

### LINKED ISSUE #1: IntelliJ shows target selection dialog incorrectly
having the following configuration
```json
{"target": {"deployment": "b"}}
```
shows dialog, since it expects path to have something while actually it can be like above.

### LINKED ISSUE #3: Multiple configuration files don't work in IntelliJ
While working with a user I noticed that creating another file ending with `mirrord.json` didn't work and didn't display the dropdown menu. We also tried restarting the IDE and it didn't help.

### LINKED ISSUE #27: Mirrord opens the config file from another open project.
### Steps to Reproduce

1. Open a project with a mirrord config file in a pycharm window.
2. Create a new project in a new PyCharm window.
3. Click on the config file symbol.
4. The config file from the other project opens in the new project.

### Your operating system and version

Linux

### LINKED ISSUE #31: Revamp configs
Need to do [the same as in the VS Code extension](https://github.com/metalbear-co/mirrord-vscode/pull/19)

### PR DESCRIPTION
**Pull request by Razz4780**
Closes #31 
Closes #27 
Closes #3
Closes #1 

Changes:
1. Dropdown menu is now always displayed next to the enable/disable button. Contains actions to enable/disable mirrord, select active config, open editor for selected config (same as in vscode ext).
2. Most notifications can now be disabled, these settings are accessible through the UI (`Settings` -> `Tools` -> `mirrord`)
3. Changed most global objects we used to classes created by a dedicated project service (it's easier to separate state between open projects this way)
4. Fixed config parsing. Plugin reads the config and checks whether target is specified. Only simple form `"target": "path"` was supported. Now the complex form ("path" as a JSON object) is supported

Sorry for a big change, but I couldn't let it go :')

EDIT: I made one change to the execution flow with mirrord enabled. When there is no available mirrord config, the plugin automatically creates a default one in the `.mirrord` dir. However, our code is called from different contexts across IntelliJ platforms. When it is called from a background thread with read lock acquired (which seems to happen only in IntelliJ IDEA), we just can't create the config file using IntelliJ's VFS (which is very handy) - we either get an exception or a deadlock. In this case I cancel the mirrord run and display a notification allowing the user to create a default config with a click. All other cases are left unchanged.

### CODE REVIEW

**Comment by infiniteregrets** [comment]
sounds like something went wrong for me while testing your code, you are doing this in the wrong thread probably

```
java.lang.Throwable: Assertion failed: Write access is allowed inside write-action only (see com.intellij.openapi.application.Application.runWriteAction())
	at com.intellij.openapi.diagnostic.Logger.assertTrue(Logger.java:254)
	at com.intellij.openapi.application.impl.ApplicationImpl.assertWriteAccessAllowed(ApplicationImpl.java:1293)
	at com.intellij.openapi.vfs.newvfs.persistent.PersistentFSImpl.processEvent(PersistentFSImpl.java:834)
	at com.intellij.openapi.vfs.newvfs.persistent.PersistentFSImpl.createChildDirectory(PersistentFSImpl.java:506)
	at com.intellij.openapi.vfs.newvfs.impl.VirtualFileSystemEntry.createChildDirectory(VirtualFileSystemEntry.java:365)
	at com.metalbear.mirrord.MirrordConfigAPI.createDefaultConfig(MirrordConfigAPI.kt:172)
	at com.metalbear.mirrord.MirrordConfigDropDown$SettingsAction$actionPerformed$2.invoke(MirrordConfigDropDown.kt:68)
	at com.metalbear.mirrord.MirrordConfigDropDown$SettingsAction$actionPerformed$2.invoke(MirrordConfigDropDown.kt:67)
	at com.metalbear.mirrord.MirrordConfigDropDown$SettingsAction.actionPerformed(MirrordConfigDropDown.kt:89)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.doPerformActionOrShowPopup(ActionUtil.java:327)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.lambda$performActionDumbAwareWithCallbacks$4(ActionUtil.java:306)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performDumbAwareWithCallbacks(ActionUtil.java:350)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAwareWithCallbacks(ActionUtil.java:306)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.invokeAction(ActionUtil.java:528)
	at com.intellij.ui.popup.ActionPopupStep.performAction(ActionPopupStep.java:252)
	at com.intellij.ui.popup.ActionPopupStep.lambda$onChosen$1(ActionPopupStep.java:229)
	at com.intellij.ui.popup.AbstractPopup.lambda$dispose$17(AbstractPopup.java:1585)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.lambda$doWhenFocusSettlesDown$3(FocusManagerImpl.java:173)
	at com.intellij.util.ui.EdtInvocationManager.invokeLaterIfNeeded(EdtInvocationManager.java:33)
	at com.intellij.ide.IdeEventQueue.ifFocusEventsInTheQueue(IdeEventQueue.java:176)
	at com.intellij.ide.IdeEventQueue.executeWhenAllFocusEventsLeftTheQueue(IdeEventQueue.java:129)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.doWhenFocusSettlesDown(FocusManagerImpl.java:169)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.doWhenFocusSettlesDown(FocusManagerImpl.java:163)
	at com.intellij.ui.popup.AbstractPopup.dispose(AbstractPopup.java:1583)
	at com.intellij.ui.popup.WizardPopup.dispose(WizardPopup.java:165)
	at com.intellij.ui.popup.list.ListPopupImpl.dispose(ListPopupImpl.java:377)
	at com.intellij.ui.popup.PopupFactoryImpl$ActionGroupPopup.dispose(PopupFactoryImpl.java:288)
	at com.intellij.openapi.util.ObjectTree.runWithTrace(ObjectTree.java:127)
	at com.intellij.openapi.util.ObjectTree.executeAll(ObjectTree.java:159)
	at com.intellij.openapi.util.Disposer.dispose(Disposer.java:243)
	at com.intellij.openapi.util.Disposer.dispose(Disposer.java:231)
	at com.intellij.ui.popup.WizardPopup.disposeAllParents(WizardPopup.java:273)
	at com.intellij.ui.popup.list.ListPopupImpl.disposePopup(ListPopupImpl.java:470)
	at com.intellij.ui.popup.list.ListPopupImpl.handleNextStep(ListPopupImpl.java:497)
	at com.intellij.ui.popup.list.ListPopupImpl._handleSelect(ListPopupImpl.java:464)
	at com.intellij.ui.popup.list.ListPopupImpl.handleSelect(ListPopupImpl.java:412)
	at com.intellij.ui.popup.PopupFactoryImpl$ActionGroupPopup.handleSelect(PopupFactoryImpl.java:300)
	at com.intellij.ui.popup.list.ListPopupImpl$MyMouseListener.mouseReleased(ListPopupImpl.java:666)
	at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:298)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6648)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3392)
	at com.intellij.ui.popup.list.ListPopupImpl$MyList.processMouseEvent(ListPopupImpl.java:731)
	at java.desktop/java.awt.Component.processEvent(Component.java:6413)
	at java.desktop/java.awt.Container.processEvent(Container.java:2266)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5022)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4854)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2802)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4854)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:781)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:730)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:724)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:754)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:752)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:751)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:909)
	at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.java:831)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:753)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$5(IdeEventQueue.java:437)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:787)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:436)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:113)
	at com.intellij.ide.IdeEventQueue.performActivity(IdeEventQueue.java:615)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:434)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:838)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:480)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)

```

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordActiveConfigWatch.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+                    service.notifier.notification(
+                        "mirrord active config has been removed",
+                        NotificationType.WARNING
+                    )
+                        .withDontShowAgain(MirrordSettingsState.NotificationId.ACTIVE_CONFIG_REMOVED)
+                        .fire()
+                }
+            }
+        } else {
+            null
```
Comment (infiniteregrets) [inline modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordActiveConfigWatch.kt:30]: small suggestion, I think for file moves you should notify the user that the config has been moved rather than removed

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt [hunk 2]
```diff
@@ -1,197 +1,138 @@
...
 
-    override fun getActionUpdateThread() = ActionUpdateThread.BGT
+            val fileManager = VirtualFileManager.getInstance()
+            val projectLocator = ProjectLocator.getInstance()
+            val configs = FileBasedIndex
+                .getInstance()
+                .getAllKeys(MirrordConfigIndex.key, service.project)
+                .mapNotNull { fileManager.findFileByUrl(it) }
+                .filter { projectLocator.getProjectsForFile(it).contains(service.project) }
+                .associateBy { it.presentableUrl }
```
Comment (infiniteregrets) [inline modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt:43]: i suggest you also consider this edge case where a directory be also named as `mirrord.json` and make sure we dont add that to our index query

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt [hunk 3]
```diff
@@ -1,197 +1,138 @@
...
-            }
-        }
-        return DefaultActionGroup().apply {
-            addAll(actions)
-        }
-    }
+            val selection = MirrordConfigDialog(
+                "Change mirrord active configuration",
+                configs.keys.toList().sorted()
+            ).show() ?: return
```
Comment (infiniteregrets) [inline modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt:48]: for shorter paths this title doesnt show up properly, maybe consider shortening it?

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt [hunk 4]
```diff
@@ -1,197 +1,138 @@
...
-            if (it.path.startsWith(basePath)) {
-                updatedConfigFiles.add(it.path)
+            val options = configs.keys.toList().sorted()
+
+            val selected = if (options.size == 1) {
+                options.first()
+            } else if (options.size > 1) {
+                MirrordConfigDialog("Edit mirrord configuration", options).show()?.option
+            } else {
+                null
```
Comment (infiniteregrets) [inline modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt:84]: kotlin suggestion, i think we can simplify this:

```kt
val options = configs.keys.toList().sorted()

val selected = options.singleOrNull()?.let { singleOption ->
    MirrordConfigDialog("Edit mirrord configuration", options).show()?.option ?: singleOption
}
```

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt (hunk 2 above)
Comment (infiniteregrets) [inline modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt:43]: i see you have not considered race conditions here/ states where we could have bad data.

querying the index in index mode is not good, we have an implementation in the main branch for this, so i suggest you change the querying to follow that approach:

```
    private fun blockAndQueryIndex(project: Project) {
        if (updateConfigs && !blockQueries) {
            blockQueries = true
            queryIndexInSmartMode(project, ::updateConfigFiles)
        }
    }
    ```

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt [hunk 5]
```diff
@@ -1,197 +1,138 @@
...
+                    FileEditorManager.getInstance(service.project).openFile(config, true)
                 }
-
-                e.presentation.text = chosenFile?.let { getReadablePath(it, project) }
             } else {
-                chosenFile = configFiles.firstOrNull()
+                configs["(default) %s".format(defaultConfig.presentableUrl)] = {
+                    FileEditorManager.getInstance(service.project).openFile(defaultConfig, true)
+                }
             }
```
Comment (infiniteregrets) [inline modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt:75]: so the Settings always shows active and default only?

**Review by infiniteregrets — CHANGES_REQUESTED** [review]
great work on the PR, done reviewing it partially, will do another soon. 
rn we need:
- fix the possibility of races/bad states
- make sure the write actions happen in the correct thread, see attached stacktrace
- add e2e

**Comment by infiniteregrets** [comment]
Looked into e2e a bit for this, one you have the xpath for the configdropdown, you can `findText` for `Settings` and the rest of the flow should remain the same!

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt (hunk 2 above)
Comment (Razz4780) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt]: I added a filter excluding directories here :)

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt (hunk 3 above)
Comment (Razz4780) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt:49]: I made the dialog wider to fit the title

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt (hunk 4 above)
Comment (Razz4780) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt]: This actually doesn't do what we want, but I changed the expression to be more "chainy"

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt (hunk 2 above)
Comment (Razz4780) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt]: Do you mean dumb mode? If so, this is already handled by the platform. The while popup is enabled only in the smart mode

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt (hunk 5 above)
Comment (Razz4780) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt:76]: Yup, at least for now

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt (hunk 2 above)
Comment (infiniteregrets) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt]: Can you please elaborate? How is handled already?

Comment (Razz4780) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt]: I think for the action to be enabled in dumb mode you have to explicitly implement the `DumbAware` interface

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordActiveConfigWatch.kt (hunk 1 above)
Comment (Razz4780) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordActiveConfigWatch.kt]: Done, I added info about new parent directory

**Comment by Razz4780** [comment]
> sounds like something went wrong for me while testing your code, you are doing this in the wrong thread probably

How did this happen? Could you specify the platform and what you were trying to do?

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt (hunk 2 above)
Comment (infiniteregrets) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt]: That’s just for that action, querying index is separate

**Comment by Razz4780** [comment]
> > > sounds like something went wrong for me while testing your code, you are doing this in the wrong thread probably
> > 
> > 
> > How did this happen? Could you specify the platform and what you were trying to do?
> 
> Just happened while normal usage while I created a default config

Should be fixed now :)

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt (hunk 2 above)
Comment (infiniteregrets) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt]: Yup

**Comment by infiniteregrets** [comment]
> > > > sounds like something went wrong for me while testing your code, you are doing this in the wrong thread probably
> > > 
> > > 
> > > How did this happen? Could you specify the platform and what you were trying to do?
> > 
> > 
> > Just happened while normal usage while I created a default config
> 
> Should be fixed now :)

Awesome! I’ll review again soon

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt (hunk 2 above)
Comment (Razz4780) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt]: After some reading - I think that's not possible, since updating indices is done in a `WriteAction` and this action has read access.

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordProjectService.kt [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        }
+
+    /**
+     * Shows a notification asking for consent to send telemetries
+     */
+    private fun telemetryConsent() {
+        ApplicationManager.getApplication().invokeLater {
+            notifier
+                .notification(
+                    "Allow mirrord to send telemetries",
```
Comment (aviramha) [inline modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordProjectService.kt:60]: I think we should rephrase it to "Allow mirrord to check for plugin updates" and leave only `versionCheckEnabled`

**Comment by aviramha** [comment]
> Allowed specifying mirrord config per launch configuration via the MIRRORD_CONFIG_FILE environment variable.
Was this done here too?

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt (hunk 2 above)
Comment (infiniteregrets) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt]: Okay so as far as I can remember update is called every half second even when the action is unavailable. So I’d say check we don’t do anything related to keeping state of configs while indexing is on

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordProjectService.kt (hunk 6 above)
Comment (Razz4780) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordProjectService.kt:60]: So always send telemetries?

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigAPI.kt [hunk 7]
```diff
@@ -35,54 +31,156 @@ object MirrordConfigAPI {
...
-        val configPath = getConfigPath(project)
-        if (!configPath.exists()) {
-            return null
+    private fun getMirrordDir(): VirtualFile? {
+        val dir = getMirrordDirParent().findChild(".mirrord") ?: return null
+
+        return if (dir.isDirectory) {
+            dir
+        } else {
+            null
```
Comment (infiniteregrets) [inline modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigAPI.kt:158]:
```
return getMirrordDirParent().findChild(".mirrord")?.takeIf { it.isDirectory }
```

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt (hunk 3 above)
Comment (infiniteregrets) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt:50]: awesome!

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordExecManager.kt [hunk 8]
```diff
@@ -79,44 +68,129 @@ object MirrordExecManager {
...
+                config = try {
+                    val path = service.configApi.getConfigPath(configFromEnv, true)
+                    Pair(path, null)
+                } catch (e: InvalidProjectException) {
+                    Pair(null, e)
+                }
+            }
+
+            config.second?.let { throw it }
+            config.first!!
```
Comment (infiniteregrets) [inline modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordExecManager.kt:118]: I am not sure I understand what this does? also we should avoid `!!`, will cause exceptions

Comment (Razz4780) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordExecManager.kt:118]: Above that I run some code in the EDT thread using `invokeAndWait`. That code may throw an exception. If it's not caught, it produces an ugly error notification in the UI. Therefore, I catch the exception and pass the result/exception back to the background thread wrapped in the `Pair<String, Exception>`. One of those is always not null, so tbh I think `!!` is ok

Comment (infiniteregrets) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordExecManager.kt:118]: can we handle this null case by any chance and propagate the error back to the user?

File: src/test/kotlin/com/metalbear/mirrord/utils/IdeaFrame.kt [hunk 9]
```diff
@@ -40,13 +41,13 @@ class IdeaFrame(remoteRobot: RemoteRobot, remoteComponent: RemoteComponent) :
         )
 
     val stopDebugging
-        get() = find<ContainerFixture>(
-            byXpath("//div[contains(@myvisibleactions, 'Me')]//div[@myaction.key='action.Stop.text']")
-        )
+        get() = findAll<ContainerFixture>(
+            byXpath("//div[@class='ActionButton' and @myaction='Stop (Stop process)']")
+        ).first()
```
Comment (infiniteregrets) [inline src/test/kotlin/com/metalbear/mirrord/utils/IdeaFrame.kt:46]: why this change?

Comment (Razz4780) [reply src/test/kotlin/com/metalbear/mirrord/utils/IdeaFrame.kt:46]: Simple `find` failed for me, I had two of those buttons - one in the toolbar, one in the sidebar of the debugging tab.

**Review by infiniteregrets — APPROVED** [review]

File: src/test/kotlin/com/metalbear/mirrord/utils/IdeaFrame.kt (hunk 9 above)
Comment (infiniteregrets) [reply src/test/kotlin/com/metalbear/mirrord/utils/IdeaFrame.kt:46]: do you mean locally?

Comment (infiniteregrets) [reply src/test/kotlin/com/metalbear/mirrord/utils/IdeaFrame.kt:46]: wait so it is the same locally and on the CI?

Comment (Razz4780) [reply src/test/kotlin/com/metalbear/mirrord/utils/IdeaFrame.kt:46]: I wanted to be prepared for more cases. There was a time that CI failed because IDE for test opened without project tree view expanded. And it has passed before

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordExecManager.kt (hunk 8 above)
Comment (Razz4780) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordExecManager.kt]: I got rid of `!!` and moved the notification out of this method

File: src/test/kotlin/com/metalbear/mirrord/utils/IdeaFrame.kt (hunk 9 above)
Comment (infiniteregrets) [reply src/test/kotlin/com/metalbear/mirrord/utils/IdeaFrame.kt:46]: I see sorry for the confusion, now it does open both on the CI and locally?

Comment (infiniteregrets) [reply src/test/kotlin/com/metalbear/mirrord/utils/IdeaFrame.kt:46]: Awesome!
